src/amber/cli/recipes/recipe.cr

```diff
@@ -0,0 +1,148 @@
+require "teeplate"
+require "liquid"
+require "base64"
+require "./filters"
+
+require "random/secure"
+require "../helpers/helpers"
+
+require "./file_entries"
+require "./recipe_fetcher"
+require "./app"
+require "./controller"
+require "./scaffold/controller"
+require "./scaffold/view"
+
+module Amber::Recipes
+  AMBER_RECIPE_FOLDER = "./.amber_recipe_cache"
+
+  class Recipe
+    getter name : String
+    getter directory : String
+    getter recipe : String
+
+    def self.can_generate?(template_type, recipe)
+      return false unless ["app", "controller", "scaffold" ].includes? template_type
+
+      template = RecipeFetcher.new(template_type, recipe).fetch
+      template.nil? ? false : true
+    end
+
+    def initialize(name : String, directory : String, recipe : String, fields = [] of String)
+      if name.match(/\A[a-zA-Z]/)
+        @name = name.underscore
+      else
+        raise "Name is not valid."
+      end
+
+      @directory = File.join(directory)
+      unless Dir.exists?(@directory)
+        Dir.mkdir_p(@directory)
+      end
+
+      @recipe = recipe
+
+      @fields = fields
+    end
+
+    def cleanup
+      # if we used a template from github then remove the temp folder
+      # this should probably be called as a command so templates are cached
+      FileUtils.rm_rf(AMBER_RECIPE_FOLDER)
+    end
+
+    def generate(template : String, options = nil)
+      case template
+      when "app"
+        if options
+          puts "Rendering App #{name} in #{directory} from #{options.r}"
```

Comment:
Should these puts statements be funneled into the logger instead? 

cc @eliasjpr

Reply:
It seems that it is being funneled thru the logger see line https://github.com/amberframework/amber/pull/710/files#diff-7129a7c5dcf68d7430b572158d3977c3R88